bitriseclient/profile.go

```diff
@@ -0,0 +1,387 @@
+	Processed      bool   `json:"processed"`
+	IsExpose       bool   `json:"is_expose"`
+	IsProtected    bool   `json:"dais_protectedta"`
+}
+
+// FetchProvisioningProfileListResponse ...
+type FetchProvisioningProfileListResponse struct {
+	Data []FetchProvisioningProfileListResponseData `json:"data"`
+}
+
+// FetchUploadedProvisioningProfileResponseData ...
+type FetchUploadedProvisioningProfileResponseData struct {
+	UploadFileName string `json:"upload_file_name"`
+	UploadFileSize int    `json:"upload_file_size"`
+	Slug           string `json:"slug"`
+	Processed      bool   `json:"processed"`
+	IsExpose       bool   `json:"is_expose"`
+	IsProtected    bool   `json:"dais_protectedta"`
+	DownloadURL    string `json:"download_url"`
+}
+
+// FetchUploadedProvisioningProfileResponse ...
+type FetchUploadedProvisioningProfileResponse struct {
+	Data FetchUploadedProvisioningProfileResponseData `json:"data"`
+}
+
+// FetchProvisioningProfiles ...
+func (client *BitriseClient) FetchProvisioningProfiles() ([]FetchProvisioningProfileListResponseData, error) {
+	log.Debugf("\nDownloading provisioning profile list from Bitrise...")
+
+	requestURL, err := urlutil.Join(baseURL, appsEndPoint, client.selectedAppSlug, provisioningProfilesEndPoint)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugf("\nRequest URL: %s", requestURL)
+
+	request, err := createRequest(http.MethodGet, requestURL, client.headers, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Response struct
+	var requestResponse FetchProvisioningProfileListResponse
+	responseStatusCode := -1
+
+	//
+	// Perform request
+	if err := retry.Times(1).Wait(5 * time.Second).Try(func(attempt uint) error {
+		body, statusCode, err := performRequest(request)
+		if err != nil {
+			log.Warnf("Attempt (%d) failed, error: %s", attempt+1, err)
+			if !strings.Contains(err.Error(), "failed to perform request") {
+				log.Warnf("Response status: %d", statusCode)
+				log.Warnf("Body: %s", string(body))
+			}
+			return err
+		}
+
+		responseStatusCode = statusCode
+
+		// Parse JSON body
+		if err := json.Unmarshal([]byte(body), &requestResponse); err != nil {
+			return fmt.Errorf("failed to unmarshal response (%s), error: %s", body, err)
+		}
+		return nil
+
+	}); err != nil {
+		return nil, err
+	}
+
+	logDebugPretty(requestResponse)
+
+	return requestResponse.Data, nil
+}
+
+// GetUploadedProvisioningProfileUUIDby ...
+func (client *BitriseClient) GetUploadedProvisioningProfileUUIDby(profileSlug string) (UUID string, err error) {
+	downloadURL, err := client.getUploadedProvisioningProfileDownloadURLBy(profileSlug)
+	if err != nil {
+		return "", err
+	}
+
+	content, err := client.downloadUploadedProvisioningProfile(downloadURL)
+	if err != nil {
+		return "", err
+	}
+
+	plistData, err := profileutil.ProvisioningProfileFromContent([]byte(content))
+	if err != nil {
+		return "", err
+	}
+
+	data, err := profileutil.NewProvisioningProfileInfo(*plistData, profileutil.ProfileTypeIos)
```

Comment:
return profileutil.NewProvisioningProfileInfo(*plistData, profileutil.ProfileTypeIos)

Reply:
We only need the UUID